tests/integration/targets/test_mysql_user/tasks/test_privs.yml

```diff
@@ -173,12 +173,62 @@
         state: present
       register: result
 
-    # FIXME: on mariadb >=10.5.2 there's always a change because the REPLICATION CLIENT privilege was renamed to BINLOG MONITOR
+    # FIXME: on mysql >=8 and mariadb >=10.5.2 there's always a change because the REPLICATION CLIENT privilege was renamed to BINLOG MONITOR
     - name: Assert that priv did not change
       assert:
         that:
           - result is not changed
-      when: install_type == 'mysql' or (install_type == 'mariadb' and mariadb_version is version('10.2', '=='))
+      when: (install_type == 'mysql' and mysql_version is version('8', '<')) or (install_type == 'mariadb' and mariadb_version is version('10.2', '=='))
+
+    # ============================================================
+    - name: grant all privileges with grant option
+      mysql_user:
+        <<: *mysql_params
+        name: '{{ user_name_2 }}'
+        password: '{{ user_password_2 }}'
+        priv: '*.*:ALL,GRANT'
+        state: present
+      register: result
+
+    - name: Assert that priv changed
+      assert:
+        that:
+          - "\"granted ['ALL', 'GRANT']\" in result.msg"
+
+    - name: Assert that 'GRANT' permission is present
+      assert:
+        that:
+          - "\"granted ['ALL', 'GRANT']\" in result.msg"
+
+    - name: Test idempotency (expect ok)
+      mysql_user:
+        <<: *mysql_params
+        name: '{{ user_name_2 }}'
+        password: '{{ user_password_2 }}'
+        priv: '*.*:ALL,GRANT'
+        state: present
+      register: result
+
+    # FIXME: on mysql >=8 and mariadb >=10.5.2 there's always a change because the REPLICATION CLIENT privilege was renamed to BINLOG MONITOR
+    - name: Assert that priv did not change
+      assert:
+        that:
+          - result is not changed
+      when: (install_type == 'mysql' and mysql_version is version('8', '<')) or (install_type == 'mariadb' and mariadb_version is version('10.2', '=='))
+
+    - name: Assert that 'GRANT' permission is present
+      assert:
+        that:
+          - "\"granted ['ALL', 'GRANT']\" in result.msg"
```

Comment:
This assertion fails. I tested with the following : mariadb 10.3.36 (I configured 10.3.34 !?), python 3.9, pymysql 0.9.3 and ansible 2.13. I debug the result var:
```json
{
    "result": {
        "changed": true,
        "failed": false,
        "msg": "Privileges updated: granted ['GRANT', 'USAGE'], revoked []",
        "password_changed": false,
        "user": "db_user2"
    }
}
```